src/dotnet/AutoRest/Plugins/AzureValidator.cs

```diff
@@ -71,6 +71,33 @@ private void LogValidationMessage(ValidationMessage validationMessage)
 
     protected override async Task<bool> ProcessInternal()
     {
+        var docStateInput = await GetValue<string>("merge-state");
+        ServiceDefinitionDocumentState docState;
+        if (!Enum.TryParse<ServiceDefinitionDocumentState>(docStateInput, true, out docState))
+        {
+            throw new Exception("Invalid Input for merge-state: " + docStateInput + ". Valid values are 'individual' and 'composed'.");
+        }
+
+        if (await GetValue<bool>("azure-validator.debugger"))
+        {
+            Debugger.Await();
+        }
+        switch (docState)
```

Comment:
nit: mixing switch statement and if condition here looks a little redundant, I'd simply use 2 if conditions (and && the predicates)